editoast/src/models/timetable.rs

```diff
@@ -63,12 +66,22 @@ impl Timetable {
         self,
         db_pool: Data<DbPool>,
     ) -> Result<TimetableWithSchedulesDetails> {
-        use crate::tables::osrd_infra_trainschedule::dsl;
+        use crate::tables::osrd_infra_infra::dsl as infra_dsl;
+        use crate::tables::osrd_infra_scenario::dsl as scenario_dsl;
+        use crate::tables::osrd_infra_trainschedule::dsl as train_dsl;
         block::<_, Result<_>>(move || {
-            let mut conn = db_pool.get()?;
-            let train_schedules: Vec<TrainSchedule> = dsl::osrd_infra_trainschedule
-                .filter(dsl::timetable_id.eq(self.id.unwrap()))
+            let mut conn = db_pool.clone().get()?;
+            let train_schedules: Vec<TrainSchedule> = train_dsl::osrd_infra_trainschedule
+                .filter(train_dsl::timetable_id.eq(self.id.unwrap()))
                 .load(&mut conn)?;
+            let infra_id: i64 = scenario_dsl::osrd_infra_scenario
+                .select(scenario_dsl::infra_id)
+                .filter(scenario_dsl::timetable_id.eq(self.id.unwrap()))
+                .first::<i64>(&mut conn)?;
+            let infra_version: String = infra_dsl::osrd_infra_infra
+                .select(infra_dsl::version)
+                .filter(infra_dsl::id.eq(infra_id))
+                .first::<String>(&mut conn)?;
```

Comment:
This could be merged into one db request with a `join` but for that you'd need to add some lines to `tables.rs`, I don't know if it's out of this PR's scope

You could even go a step further and use [diesel's relations](https://diesel.rs/guides/relations.html) to have very pretty code